File: app/src/main/java/com/future/awaker/email/MyAuthenticator.java
package com.future.awaker.email;

import javax.mail.Authenticator;
import javax.mail.PasswordAuthentication;

/**
 * Copyright ©2017 by ruzhan
 * <p>
 * 认证类
 */

public class MyAuthenticator extends Authenticator {

    String userName = null;
    String password = null;

    public MyAuthenticator() {
    }

    public MyAuthenticator(String username, String password) {
        this.userName = username;
        this.password = password;
    }

    protected PasswordAuthentication getPasswordAuthentication() {
        return new PasswordAuthentication(userName, password);
    }
}

File: app/src/main/java/com/future/awaker/email/MyAuthenticator.kt
package com.future.awaker.email

import javax.mail.Authenticator
import javax.mail.PasswordAuthentication


class MyAuthenticator : Authenticator {

    internal var userName: String? = null
    internal var password: String? = null

    constructor() {}

    constructor(username: String, password: String) {
        this.userName = username
        this.password = password
    }

    override fun getPasswordAuthentication(): PasswordAuthentication {
        return PasswordAuthentication(userName, password)
    }
}
